cmake/OpenCVCompilerOptions.cmake

```diff
@@ -377,10 +377,15 @@ if(MSVC)
 endif()
 
 if(NOT OPENCV_FP16_DISABLE)
+  if(ARM)
+    set(FP16_OPTION "-mfpu=neon-fp16")
+  elseif(X86 OR X86_64)
+    set(FP16_OPTION "-mf16c")
+  endif()
```

Comment:
Thank you @alalek for your time and reviewing my PR.
I updated the PR and I had to change this ```try_compile``` to ```try_run```
I'd like to hear your opinion about that point.

Best.